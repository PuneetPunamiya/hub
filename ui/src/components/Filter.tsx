import React from 'react'
import {Checkbox} from '@patternfly/react-core'
import {useObserver} from 'mobx-react'

interface store {
	store: any
}

const Filter: React.FC<store> = (props: store) => {

	return useObserver(() => (
		<div>
			{
				props.store.map((c: any) => {
					return (
						<Checkbox
							key={c.id}
							label={c.name}
							isChecked={c.selected}
							onChange={() => c.toggle()}
							aria-label="controlled checkbox example"
							id="store data"
							name={c.name}
						/>
					)
				})
			}

		</div>
	))
}

export default Filter
