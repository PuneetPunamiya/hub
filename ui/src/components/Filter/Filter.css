.Filter {
  margin: 3em;
}

.Text-Header {
  width: 10em;
}

.Checkboxes {
  padding-left: 0.7em;
}
