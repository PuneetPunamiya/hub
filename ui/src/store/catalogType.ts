import { types } from "mobx-state-tree";
export const CatalogTypeList = [
  { name: "official", selected: false },
  { name: "verified", selected: false },
  { name: "community", selected: false }
];

const Type = types
  .model({
    name: types.string,
    selected: false
  })
  .actions(self => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export const CatalogTypeStore = types
  .model({
    catalogtypelist: types.array(Type)
  })
  .views(self => ({
    get count() {
      return self.catalogtypelist.length;
    },
    get catalogType() {
      const { catalogtypelist } = self;
      return catalogtypelist
        .filter((type: any) => type.selected === true)
        .map((type: any) => type.name);
    }
  }))
  .actions(self => ({
    add(item: any) {
      self.catalogtypelist.push(item);
    },
    setSelectedCatalogType(kindName: any) {
      self.catalogtypelist.forEach((kind: any) => {
        if (kind.name === kindName) {
          kind.toggle();
          return;
        }
      });
    }
  }))
  .actions(self => ({
    loadCatalogType() {
      CatalogTypeList.forEach((kind: any) => self.add(kind));
    }
  }))
  .actions(self => ({
    afterCreate() {
      self.loadCatalogType();
    }
  }));
