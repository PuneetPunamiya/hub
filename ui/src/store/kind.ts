import { types } from "mobx-state-tree";

const Kind = types
	.model({
		name: types.string,
		selected: false,
	})
	.actions((self) => ({
		toggle() {
			self.selected = !self.selected;
		},
	}));

export const KindStore = types
	.model({
		kind: types.array(Kind),
	})
	.views((self) => ({
		get count() {
			return self.kind.length;
		},
	}))
	.actions((self) => ({
		add(item: any) {
			self.kind.push(item);
		},
	}));
