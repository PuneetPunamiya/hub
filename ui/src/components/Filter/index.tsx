import React from 'react';
import { useObserver } from 'mobx-react';
import { Button, Checkbox, Text, TextVariants, Grid, GridItem } from '@patternfly/react-core';
import TimesIcon from '@patternfly/react-icons/dist/js/icons/times-icon';
import './Filter.css';

interface Filterable {
  id?: number;
  name: string;
  selected: boolean;
  toggle(): void;
}

interface Store {
  values: Filterable[];
  clearSelected(): void;
}

interface Props {
  store: Store;
  header: string;
}

const checkboxes = (values: Filterable[]) => {
  return values.map((c: Filterable) => (
    <Checkbox
      key={c.name}
      label={c.name}
      isChecked={c.selected}
      onChange={() => c.toggle()}
      aria-label="controlled checkbox"
      id={`${c.id}`}
      name={c.name}
    />
  ));
};

const Filter: React.FC<Props> = ({ store, header }) => {
  return useObserver(() => (
    <div>
      <Grid>
        <GridItem span={6}>
          <Text component={TextVariants.h1} className="hub-filter-header">
            {header}
          </Text>
        </GridItem>
        <GridItem span={3}>
          <Button variant="plain" aria-label="Clear" onClick={store.clearSelected}>
            <TimesIcon />
          </Button>
        </GridItem>

        <GridItem span={12}>{checkboxes(store.values)}</GridItem>
      </Grid>
    </div>
  ));
};

export default Filter;
