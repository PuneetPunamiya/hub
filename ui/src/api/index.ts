import axios from "axios";
import { API_URL } from "../config/constants";
import { ICategory } from "../store/category";
import { IResource } from "../store/resources";
export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
}
export class Hub implements Api {
  async resources() {
    try {
      return axios.get(
        `https://api-tekton-hub-staging.apps.openshift-web.p0s5.p1.openshiftapps.com/resources`
      );
    } catch (err) {
      return err.response;
    }
  }
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`);
    } catch (err) {
      return err.response;
    }
  }
}
