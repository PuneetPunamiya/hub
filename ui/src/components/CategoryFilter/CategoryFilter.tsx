import React from 'react'
import {useObserver} from "mobx-react";
import {Flex, FlexItem} from '@patternfly/react-core';
import {
	Button,
} from '@patternfly/react-core/dist/js/components';
import TimesIcon from '@patternfly/react-icons/dist/js/icons/times-icon';
import Filter from '../Filter';
import {ICategoryStore} from '../../store/category';

interface store {
	store: ICategoryStore
}

const CategoryFilter: React.FC<store> = (props: store) => {

	return useObserver(() => (
		<div className="CategoryFilter" style={{margin: '5em'}}>
			<h2>Categories</h2>

			<Flex>
				<FlexItem>
					<Filter store={props.store.categories} />
				</FlexItem>
				<FlexItem>
					<Button variant="plain" aria-label="Action" onClick={props.store.clearAll}>
						<TimesIcon />
					</Button><br /><br />
				</FlexItem>
			</Flex>

		</div>
	))

}

export default CategoryFilter


