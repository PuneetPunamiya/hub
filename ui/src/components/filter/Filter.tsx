import React from 'react'
import {Checkbox} from '@patternfly/react-core'
import {useObserver} from 'mobx-react'
import {ICategory} from '../../store/category'

interface store {
	store: ICategory[]
}

const list = (props: store) => {
	return props.store.map((c: ICategory) => {
		console.log(props.store)
		return (
			<Checkbox
				key={c.id}
				label={c.name}
				isChecked={c.selected}
				onChange={() => c.toggle()}
				aria-label="controlled checkbox example"
				id="store-data"
				name={c.name}
			/>
		)
	})
}

const Filter: React.FC<store> = (props: store) => {

	return useObserver(() => (
		<div>
			{list(props)}
		</div>
	))
}

export default Filter
