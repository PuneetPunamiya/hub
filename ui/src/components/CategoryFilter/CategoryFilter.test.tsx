import React from 'react';
import {shallow} from 'enzyme';
import {FakeHub} from '../../api/testutil';
import {CategoryStore} from '../../store/category';
import CategoryFilter from '../CategoryFilter/CategoryFilter';
import Filter from '../Filter';

const TESTDATA_DIR = `${__dirname}/testdata`;
const api = new FakeHub(TESTDATA_DIR);

it("matches snapshot before and after loading", (done) => {
	const store = CategoryStore.create({}, {api});

	const component = shallow(<CategoryFilter store={store} />)

	expect(component.find(Filter).length).toEqual(1)
	expect(component).toMatchSnapshot()

	done()

});
