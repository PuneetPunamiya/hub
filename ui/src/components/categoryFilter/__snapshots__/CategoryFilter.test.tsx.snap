// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot before and after loading 1`] = `ShallowWrapper {}`;
